File: BEmuJava/bemu/src/main/java/com/bloomberglp/blpapi/HistoricMessage.java
//------------------------------------------------------------------------------
// <copyright project="BEmu_maven" file="/BEmu_maven/bemu/src/main/java/com/bloomberglp/blpapi/HistoricMessage.java" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

package com.bloomberglp.blpapi;

import com.bloomberglp.blpapi.Element;
import com.bloomberglp.blpapi.Name;
import com.bloomberglp.blpapi.Message;
import com.bloomberglp.blpapi.CorrelationID;
import com.bloomberglp.blpapi.Datetime;
import java.util.Map;
import java.util.List;

public class HistoricMessage extends Message
{
    private final HistoricElementSecurityData _security;
    
    HistoricMessage(CorrelationID corr, String securityName, List<String> badFields, Map<Datetime, Map<String, Object>> fieldData, int sequenceNumber) throws Exception
    {
    	super(new Name("HistoricalDataResponse"), corr, null);
    	this._security = new HistoricElementSecurityData(securityName, badFields, fieldData, sequenceNumber);
    }
    
    HistoricElementSecurityData security()
    {
    	return this._security;
    }
    
	public Element getElement(String name) throws Exception
	{
		if(name.equals(this._security.name().toString()))
			return this._security;
		else
			throw new Exception("not implemented. case-sensitive.");
	}
    
    public String toString()
    {
        StringBuilder result = new StringBuilder();
        
        result.append("HistoricalDataResponse (choice) = {" + System.getProperty("line.separator"));
        try {
			result.append(this._security.prettyPrint(1));
		} catch (Exception e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
        result.append("}");

        return result.toString();
    }
}

File: BEmu/HistoricalDataRequest/HistoricMessage.cs
//------------------------------------------------------------------------------
// <copyright project="BEmu_csh" file="HistoricalDataRequest/HistoricMessage.cs" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

namespace Bloomberglp.Blpapi.HistoricalDataRequest
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;

    internal class HistoricMessage : Message
    {
        private readonly HistoricElementSecurityData _security;

        internal HistoricMessage(CorrelationID corr, string securityName, List<string> badFields, Dictionary<DateTime, Dictionary<string, object>> fieldData, int sequenceNumber)
            : base(new Name("HistoricalDataResponse"), corr, null)
        {
            this._security = new HistoricElementSecurityData(securityName, badFields, fieldData, sequenceNumber);
        }

        public override IEnumerable<Element> Elements { get { yield return this._security; } }
        public override object this[string name, int index] { get { return null; } }
        public override Element AsElement { get { return new HistoricElementReference(this); } }

        public override string ToString()
        {
            StringBuilder result = new StringBuilder();
            result.Append("HistoricalDataResponse (choice) = {" + Environment.NewLine);
            result.Append(this._security.PrettyPrint(1));
            result.Append("}");

            return result.ToString();
        }

    }
}
